CMakeLists.txt

```diff
@@ -54,6 +54,9 @@ option(CUB_ENABLE_THOROUGH_TESTING "Build CUB thorough test variants." ON)
 option(CUB_ENABLE_BENCHMARK_TESTING  "Build CUB benchmark test variants." ON)
 option(CUB_ENABLE_MINIMAL_TESTING "Build CUB minimal test variants." ON)
 option(CUB_ENABLE_EXAMPLES "Build CUB examples." ON)
+option(CUB_NVCXX_QA_COMPAT "Use shorter names and force a single dialect." OFF)
```

Comment:
I'd prefer a name that describes what it does, not what it is. I'd suggest something like `CUB_ENABLE_CPP_DIALECT_IN_EXE_NAMES` (defaulted to `ON`). I don't think the name needs to capture the fact that you get a configuration error if you try to use it with multiple C++ dialects; that just seems like a natural consequence.